PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -1139,6 +1188,169 @@ void PMTraceConsumer::HandleDWMEvent(EVENT_RECORD* pEventRecord)
+
+    // mPresentByThreadId
+    {
+        auto eventIter = mPresentByThreadId.find(p->ThreadId);
+        if (eventIter != mPresentByThreadId.end() && eventIter->second == p) {
+            mPresentByThreadId.erase(eventIter);
+        }
+        // TODO: Assert that an element has been removed if this element must exist.
+    }
+
+    // mPresentsBySubmitSequence
+    {
+        // TODO: is 0 a valid QueueSubmitSequence?
+        if (p->QueueSubmitSequence != 0) {
+            auto eventIter = mPresentsBySubmitSequence.find(p->QueueSubmitSequence);
+            if (eventIter != mPresentsBySubmitSequence.end() && (eventIter->second == p)) {
+                mPresentsBySubmitSequence.erase(eventIter);
+                // TODO: Assert elementsRemoved = 1 if this element must exist.
+            }
+        }
+    }
+
+    // mWin32KPresentHistoryTokens
+    {
+        if (p->ValidHistoryTokenKey) {
+            PMTraceConsumer::Win32KPresentHistoryTokenKey key(
+                p->CompositionSurfaceLuid,
+                p->Win32KPresentCount,
+                p->Win32KBindId
+            );
+
+            auto eventIter = mWin32KPresentHistoryTokens.find(key);
+            if (eventIter != mWin32KPresentHistoryTokens.end() && (eventIter->second == p)) {
+                mWin32KPresentHistoryTokens.erase(eventIter);
+                // TODO: Assert elementsRemoved = 1 if this element must exist.
+            }
+        }
+    }
+
+    // mDxgKrnlPresentHistoryTokens
+    {
+        if (p->TokenPtr != 0) {
+            auto eventIter = mDxgKrnlPresentHistoryTokens.find(p->TokenPtr);
+            if (eventIter != mDxgKrnlPresentHistoryTokens.end() && eventIter->second == p) {
+                mDxgKrnlPresentHistoryTokens.erase(eventIter);
+                // TODO: Assert that an element has been removed if this element must exist.
+            }
+        }
+    }
+
+    // mBltsByDxgContext
+    {
+        if (p->ValidHContext) {
+            auto eventIter = mBltsByDxgContext.find(p->DxgKrnlHContext);
+            if (eventIter != mBltsByDxgContext.end() && eventIter->second == p) {
+                mBltsByDxgContext.erase(eventIter);
+                // TODO: Assert that an element has been removed if this element must exist.
+            }
+        }
+    }
+
+    // mLastWindowPresent
+    {
+        // 0 is a invalid hwnd
+        if (p->Hwnd != 0) {
+            auto eventIter = mLastWindowPresent.find(p->Hwnd);
+            if (eventIter != mLastWindowPresent.end() && eventIter->second == p) {
+                mLastWindowPresent.erase(eventIter);
+            }
+            // There's no hard guaruntee that this element must exist, no assert needed here.
+        }
+    }
+
+    // mPresentsWaitingForDWM
+    {
+        if (p->WaitingForDwmCompletion)
+        {
+            auto eventIter = mPresentsWaitingForDWM.begin();
+            while (eventIter != mPresentsWaitingForDWM.end() && p != *eventIter)
+            {

```

Comment:
Nitpick: Style inconsistency. Most of the code here puts open braces on the same line. Let's stick with that.